share/doc/bdiana.1.in
.\"                                      Hey, EMACS: -*- nroff -*-
.Dd Oct 17, 2017
.Os
.Dt BDIANA 1
.Sh NAME
.Nm bdiana-@PVERSION@
.Nd MET Norway meteorological production software
.Sh SYNOPSIS
.Nm
[\fB\-v\fR]
[\fB\-L\fR \fIlog4cpp.properties\fR]
[\fB\-s\fR \fIsetupfile\fR]
[\fB\-i\fR \fIjobfile\fR]
.Sh DESCRIPTION
.Nm
is a meteorological production software developed by MET Norway, SMHI,
and others.

Please see the Diana wiki for help.

.Sh SEE ALSO
https://github.com/metno/diana, https://wiki.met.no/diana/start

.Sh AUTHOR
Diana is written at Met.no, the Norwegian Meteorological Institute,
and SMHI, the Swedish Meteorological and Hydrological Institute.  This
manual page was written by Alexander B\(:urger
<alexander.buerger@met.no> in October 2017.


debian/changelog
diana (3.28.0-ubuntu1) lucid; urgency=low

  * using fimex 0.37.1
  * using metlibs_trunk
  * compiled without profet support
	
 -- Lisbeth Bergholt <lisbethb@met.no>  Thu, 29 Mar 2012 21:42:51 +0200

diana (3.27.4-ubuntu1) lucid; urgency=low

  * Observations - bugfix criterias

 -- Lisbeth Bergholt <lisbethb@met.no>  Thu, 01 Mar 2012 18:55:51 +0100

diana (3.27.3-ubuntu1) lucid; urgency=low

  * HQC - bugfix

 -- Lisbeth Bergholt <lisbethb@met.no>  Mon, 27 Feb 2012 14:31:58 +0100

diana (3.27.2-ubuntu1) lucid; urgency=low

  * New EPS parameter - cluster representative
  * Bugfix profet: computed fields (like FF) were not
	changed when the original fields (u,v) were edited. 

 -- Lisbeth Bergholt <lisbethb@met.no>  Tue, 14 Feb 2012 20:32:42 +0100

diana (3.27.1-ubuntu1) lucid; urgency=low

  * Minor fixes

 -- Lisbeth Bergholt <lisbethb@met.no>  Thu, 09 Feb 2012 14:48:14 +0100

diana (3.27.0-ubuntu1) lucid; urgency=low

  * Using metlibs_4.3
  * Using libfimex-0.35
  * Using WDB/fimex
  * Reading from KDVH

 -- Lisbeth Bergholt <lisbethb@met.no>  Thu, 26 Jan 2012 09:42:51 +0100

diana (3.26.4-ubuntu3) lucid; urgency=low

  * bdiana: default canvas changed to QGLPixelBuffer

 -- Lisbeth Bergholt <lisbethb@pc2982.pc.met.no>  Mon, 28 Nov 2011 09:42:43 +0100

diana (3.26.4-ubuntu2) lucid; urgency=low

  * using +towgs in all proj strings except old mitiff files.
  * Using metlibs_4.2
  * Using libfimex-0.33-1

 -- Lisbeth Bergholt <lisbethb@vpn-116-99.met.no>  Sat, 26 Nov 2011 16:42:14 +0100

diana (3.26.4-ubuntu1) lucid; urgency=low

  * Bugfix:DifferenceField - plot string was not parsed correctly
  * Using metlibs_trunk (by mistake)

 -- Lisbeth Bergholt <lisbethb@met.no>  Tue, 22 Nov 2011 13:54:05 +0100

diana (3.26.3-ubuntu1) lucid; urgency=low

  * Bugfix: Ascii-observations - only observations from one file were used
  * Using metlibs_trunk (by mistake)

 -- Lisbeth Bergholt <lisbethb@.met.no>  Mon, 21 Nov 2011 13:30:46 +0100

diana (3.26.2-ubuntu1) lucid; urgency=low

  * Bugfix: mark measurements/trajectory positions

 -- Lisbeth Bergholt <lisbethb@met.no>  Sun, 09 Oct 2011 17:28:50 +0200

diana (3.26.1-ubuntu2) lucid; urgency=low

  * Bugfix bdiana: vertical levels

 -- Lisbeth Bergholt <lisbethb@met.no>  Tue, 27 Sep 2011 14:09:07 +0200

diana (3.26.1-ubuntu1) lucid; urgency=low

  * Bugfix bdiana: translation from plot name to field name

 -- Lisbeth Bergholt <lisbethb@met.no>  Tue, 27 Sep 2011 12:52:03 +0200

diana (3.26.0-ubuntu2) lucid; urgency=low

  * Projections: datum fix removed

 -- Lisbeth Bergholt <lisbethb@met.no>  Mon, 26 Sep 2011 10:39:04 +0200

diana (3.26.0-ubuntu1) lucid; urgency=low

  * Using metlibs_4.2
  * Using libfimex-0.31-2
  * New profet function
  * Improved fimex reading
  * Vertical crossections: vt+omega / vt+w fixed
  * Bufr: skip time variables with missing valaues 
  * Bugfix:projections

 -- Lisbeth Bergholt <lisbethb@met.no>  Fri, 23 Sep 2011 15:38:14 +0200

diana (3.25.5-ubuntu2) lucid; urgency=low

  * Bugfix: bdiana did not find fieldTimes

 -- Lisbeth Bergholt <lisbethb@met.no>  Mon, 04 Jul 2011 12:41:27 +0200

diana (3.25.5-ubuntu1) lucid; urgency=low

  * Bugfix: Labels from ObsAscii files
  * Using metlibs_4.1
  * Using libfimex-dev 0.29

 -- Lisbeth Bergholt <lisbethb@met.no>  Mon, 27 Jun 2011 09:45:07 +0200

diana (3.25.4-ubuntu2) lucid; urgency=low

  * Bugfix was not included in last version

 -- Lisbeth Bergholt <lisbethb@met.no>  Wed, 22 Jun 2011 08:52:03 +0200

diana (3.25.4-ubuntu1) lucid; urgency=low

  * Bugfix: Memory leak in StationPlot
  * PostScript fix: Always set font-specification

 -- Lisbeth Bergholt <lisbethb@met.no>  Tue, 21 Jun 2011 14:53:45 +0200

diana (3.25.3-ubuntu1) lucid; urgency=low

  * More bugfix: profet

 -- Lisbeth Bergholt <lisbethb@met.no>  Mon, 20 Jun 2011 13:30:54 +0200

diana (3.25.2-ubuntu2) lucid; urgency=low

  * Bugfix case insensitivity

 -- Lisbeth Bergholt <lisbethb@met.no>  Fri, 17 Jun 2011 14:54:36 +0200

diana (3.25.2-ubuntu1) lucid; urgency=low

  * Bugfix xylimit
  * Reverted fix: model and field are now case insensitive

 -- Lisbeth Bergholt <lisbethb@met.no>  Fri, 17 Jun 2011 08:28:03 +0200

diana (3.25.1-ubuntu1) lucid; urgency=low

  * Field: model and field are now case sensitive

 -- Lisbeth Bergholt <lisbethb@met.no>  Thu, 16 Jun 2011 11:26:45 +0200

diana (3.25.0-ubuntu4) lucid; urgency=low

  * Using metlibs_4.1
  * Using libfimex-dev 0.28-2

 -- Lisbeth Bergholt <lisbethb@met.no>  Wed, 15 Jun 2011 10:06:40 +0200

diana (3.25.0-ubuntu3) lucid; urgency=low

  * Using metlibs_4.1
  * Using libfimex-dev 0.28

 -- Lisbeth Bergholt <lisbethb@met.no>  Tue, 10 May 2011 08:31:50 +0200

diana (3.25.0-ubuntu2) lucid; urgency=low

  * Geotiff support enabled
  * Using metlibs_4.1

 -- Lisbeth Bergholt <lisbethb@met.no>  Thu, 05 May 2011 12:02:30 +0200

diana (3.25.0-ubuntu1) lucid; urgency=low

  * New field reader using fimex
  * Using metlibs_2011-01-28

 -- Lisbeth Bergholt <lisbethb@met.no>  Fri, 28 Jan 2011 12:47:24 +0100

diana (3.24.0-ubuntu2) lucid; urgency=low

  * Bugfix: enable/disable ObjectPlots
  * Using metlibs_2011-01-13

 -- Lisbeth Bergholt <lisbethb@met.no>  Mon, 24 Jan 2011 11:01:52 +0100

diana (3.24.0-ubuntu1) lucid; urgency=low

  * Using metlibs_2011-01-13
  * new version of profet

 -- Lisbeth Bergholt <lisbethb@met.no>  Fri, 21 Jan 2011 13:26:08 +0100

diana (3.23.3-ubuntu1) lucid; urgency=low

  * Bugfix parsing logfile
  * Using metlibs_2010-11-15

 -- Lisbeth Bergholt <lisbethb@vpn-117-63.met.no>  Mon, 06 Dec 2010 09:00:58 +0100

diana (3.23.2-ubuntu1) lucid; urgency=low

  * Bugfix obsplot: pressure tendency, temperature precision, metar clouds 
  * Bugfix editing: Snow_showers -> Snow showers

 -- Lisbeth Bergholt <lisbethb@met.no>  Mon, 29 Nov 2010 13:22:29 +0100

diana (3.23.1-ubuntu6) lucid; urgency=low

  * Added export LANG=en_US in diana.in

 -- Lisbeth Bergholt <lisbethb@met.no>  Tue, 23 Nov 2010 14:00:45 +0100


perl/Metno-Bdiana/lib/Metno/Bdiana.pm
package Metno::Bdiana;

use 5.008008;
use strict;
use warnings;

require Exporter;

our @ISA = qw(Exporter);

# Items to export into callers namespace by default. Note: do not export
# names by default without a very good reason. Use EXPORT_OK instead.
# Do not simply export all your public functions/methods/constants.

# This allows declaration	use Metno::Bdiana ':all';
# If you do not need this, moving things directly into @EXPORT or @EXPORT_OK
# will save memory.
our %EXPORT_TAGS = ( 'all' => [ qw(
	readSetupFile
    parseAndProcessString
    DI_OK
    DI_ERROR
) ] );

our @EXPORT_OK = ( @{ $EXPORT_TAGS{'all'} } );

our @EXPORT = qw(
	
);

our $VERSION = '3.33';

use constant DI_OK => 1;
use constant OK => DI_OK(); # compatibility with old version
use constant DI_ERROR => 99;
use constant ERROR => DI_ERROR(); # compatibility with old version

require XSLoader;
XSLoader::load('Metno::Bdiana', $VERSION);

# Preloaded methods go here.

my $isInit = 0;
sub init {
    deInit_();
    my $ret = init_();
    if ($ret == DI_OK) {
         $isInit++;
    }
    return $ret;
}

sub deInit_ {
    if ($isInit) {
        free();
        $isInit = 0;
    }
}

END {
    deInit_();
}

1;
__END__

=head1 NAME

Metno::Bdiana - Perl version of met.no's DIANA

=head1 SYNOPSIS

  use Metno::Bdiana qw(:all);

  Metno::Bdiana::init(); # required once

  my $setup = "/disk1/WMS/usr/share/metno-wmsservice/verportal/bdiana/diana.setup";
  readSetupFile($setup) == DI_OK or die "cannot setup diana with $setup";

  my $plot =<<'EOT';
output = PNG
colour = COLOUR
buffersize = 256x256
settime = 2012-09-13 13:00:00
addhour = 00
filename = /tmp/test.png
PLOT
MAP area=EPSG:3575 backcolour=255:255:255 xylimit=-640000,-320000,-2880000,-2560000

FIELD Proff_default NEDBOR.1T alpha=160 base=0 colour=off colour_2=off extreme.radius=1 extreme.size=1 extreme.type=Ingen field.smooth=0 grid.lines=0 grid.lines.max=0 label.size=1 line.interval=40 line.smooth=0 value.label=0  line.values=0.1,0.2,0.5,1,2,4,6,10,15,20,25,30,...100 linetype=solid linewidth=1 maxvalue=off minvalue=off palettecolours=vp_nedbor patterns=off recursive=0 table=0 undef.colour=white undef.linetype=solid undef.linewidth=1 undef.masking=0 value.label=1


ENDPLOT
EOT

  parseAndProcessString($plot) == DI_OK or die "cannot create plot";



=head1 DESCRIPTION

Metno::Bdiana gives perl-library access to the batch-diana functionality. The communication with diana follows with dianas own plotting syntax and works by simly
sending a plot-string to the parseAndProcessString function.

=head2 FUNCTIONS

=over 8

=item init()

must be called at least once before any other bdiana function. Currently, no options possible.

=item readSetupFile($filename)

parses the setupFile given in the filename string.

=item parseAndProcessString($string) 

create a plot using a diana-plot commando as string.


=back

=head2 EXPORT

None by default. ':all' gives 	readSetupFile, parseAndProcessString and the constants
DI_OK and DI_ERROR


=head1 SEE ALSO

L<diana>

=head1 AUTHOR

Heiko Klein, E<lt>heiko.klein@met.noE<gt>

=head1 COPYRIGHT AND LICENSE

Copyright (C) 2012 by Heiko Klein

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself, either Perl version 5.14.2 or,
at your option, any later version of Perl 5 you may have available.


=cut
